build.assets/Dockerfile-centos7-fips

```diff
@@ -79,6 +84,37 @@ ENV GOEXPERIMENT=boringcrypto \
     GOROOT="/opt/go" \
     PATH="/opt/llvm/bin:$PATH:/opt/go/bin:/go/bin:/go/src/github.com/gravitational/teleport/build"
 
+# Install Rust
+ARG RUST_VERSION
+ENV RUSTUP_HOME=/usr/local/rustup \
+     CARGO_HOME=/usr/local/cargo \
+     PATH=/usr/local/cargo/bin:$PATH \
+     RUST_VERSION=$RUST_VERSION
+
+RUN mkdir -p $RUSTUP_HOME && chmod a+w $RUSTUP_HOME && \
+    mkdir -p $CARGO_HOME/registry && chmod -R a+w $CARGO_HOME
+
+
+ARG LLVM_VERSION=7.0.1
+RUN mkdir -p /opt && cd /opt && curl -L https://releases.llvm.org/7.0.1/clang+llvm-$LLVM_VERSION-x86_64-linux-gnu-ubuntu-18.04.tar.xz | tar xJ
+ENV PATH="$PATH:/opt/clang+llvm-$LLVM_VERSION-x86_64-linux-gnu-ubuntu-18.04/bin"
+
+RUN ln -s /usr/bin/cmake3 /usr/bin/cmake
+
+# Build FIPS certified BoringSSL library
+# https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/3678
+ARG BORINGSSL_VERSION=ae223d6138807a13006342edfeef32e813246b39
+RUN mkdir -p /opt && cd /opt && \
+    git clone https://github.com/google/boringssl.git && \
+    cd boringssl && \
+    git checkout $BORINGSSL_VERSION && \
+    mkdir build && \
+    cd build && \
+    scl enable devtoolset-7 llvm-toolset-7.0 "cd /opt/boringssl/build && cmake -DCMAKE_C_COMPILER=clang -DFIPS=1 -DCMAKE_BUILD_TYPE=Release -GNinja .. && ninja"
```

Comment:
This is confusing and probably incorrect. First, you download the LLVM from llvm.org, but then you use `llvm-toolset` that comes from the RedHat repository, Second, you only set C compiler. I think you should also set CXX explicitly to make sure that C++ code is built with Clang too.

Reply:
I simplified the Dockerfile for CentOS so it no longer downloads the LLVM from the llvm.org (it was wrong anyway). We can fetch LLVM 7.0.1 from the CentOS repositories. I've also simplified the cmake invocation so I don't have to specify the C compiler.